dds/DCPS/RTPS/ParameterListConverter.cpp

```diff
@@ -747,6 +742,15 @@ bool from_param_list(const ParameterList& param_list,
 
 // OpenDDS::DCPS::DiscoveredWriterData
 
+void add_DataRepresentationQos(ParameterList& param_list, const DDS::DataRepresentationIdSeq& ids, bool reader)
+{
+  DDS::DataRepresentationQosPolicy dr_qos;
+  dr_qos.value = DCPS::get_effective_data_rep_qos(ids, reader);
+  Parameter param;
+  param.representation(dr_qos);
+  add_param(param_list, param);
```

Comment:
As with most other parameters, if the value is the "network message default" then the parameter should be omitted.  

In this case we have the added complexity that the "network message default" is different from the default QoS (empty sequence) and also different from the interpretation of the empty sequence.

But that shouldn't prevent this from using the common pattern of "if not (network message) default then add the parameter."

Reply:
So you're saying we should omit the parameter if we would otherwise be sending just XCDR1?

Reply:
Yeah, I guess that's a shorter way of putting it.  Hopefully the context above is helpful.

Reply:
Yes, I have no problems with that.

Reply:
@iguessthislldo @mitza-oci  I'm sure what I need to do on this.